js/index.js

```diff
@@ -0,0 +1,97 @@
+        console.log(expectedWord)
+        let attemptedWord = []
+        let characterCounter = 0
+// let attemptCounter = 0 - useful later
+
+// MAKE ON SCREEN LETTERS WORK
+        const targetOnScreenLetters = document.querySelectorAll('.key')
+        const targetBackspace = document.querySelector('.backspace')
+        const targetEnter = document.querySelector('.enter')
+        targetOnScreenLetters.forEach(function (letter) {
+            letter.addEventListener('click', function () {
+                if (attemptedWord.length < 5) {
+                    attemptedWord.push(letter.innerHTML)
+                    document.querySelector('.tile' + characterCounter).innerHTML = letter.innerHTML
+                    characterCounter++
+                    console.log(attemptedWord, characterCounter)
+                }
+            })
+        })
+
+        targetBackspace.addEventListener('click', deleteLetter)
+        targetEnter.addEventListener('click', enterPressed)
+
+        function deleteLetter () {
+            if (characterCounter > 0) {
+                attemptedWord = attemptedWord.slice(0, attemptedWord.length - 1)
+                characterCounter--
+                document.querySelector('.tile' + characterCounter).innerHTML = ''
+                // characterCounter--
+                console.log(attemptedWord, characterCounter)
+            }
+        }
+
+        function enterPressed () {
+            if (attemptedWord.length === 5) {
+                const result = checkWord(expectedWord,attemptedWord)
+                alert(result)
+            }
+        }
+
+// MAKE THE REAL KEYBOARD WORK
+        const characterSet = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'
+        document.addEventListener('keyup', (event) => {
+            if (attemptedWord.length < 5) {
+                if (characterSet.includes(event.key)) {
+                    attemptedWord.push(event.key.toUpperCase())
+                    document.querySelector('.tile' + characterCounter).innerHTML = event.key
+                    characterCounter++
+                    console.log(attemptedWord, characterCounter)
+                }
+            }
+
+            if (event.key == 'Backspace') {
+                deleteLetter()
+            }
+
+            if (event.key == 'Enter') {
+                enterPressed()
+            }
+        })
+
+        function checkWord(expectedWord,attemptedWord) {
+            let correctLetters = 0
+            let correctPositions = 0
+            for (let i = 0; i < attemptedWord.length; i++) {
+                if (attemptedWord[i] === expectedWord[i]) {
+                    correctPositions++
+                }
+                if (expectedWord.includes(attemptedWord[i]) && attemptedWord[i] !== expectedWord[i]){
+                    correctLetters++
+                }
+            }
+            return correctPositions === 5 ? 'Correct' : 'Incorrect'
```

Comment:
I would advise making this function return a boolean instead of the actual strings, then you can check it when you call it and output correct or incorrect.

Makes it easier to change the output messages without needing to muck around in this function

Reply:
done